pypeit/scripts/ql.py

```diff
@@ -0,0 +1,384 @@
+        parser.add_argument("--clobber_calibs", default=False, 
+                            action="store_true",
+                            help='Clobber existing calibration files?')
+        parser.add_argument('--slitspatnum', type=str,
+                            help='Reduce the slit(s) as specified by the slitspatnum value(s)')
+        parser.add_argument('--maskID', type=int,
+                            help='Reduce the slit(s) as specified by the maskID value(s)')
+        parser.add_argument('--boxcar_radius', type=float,
+                            help='Set the radius for the boxcar extraction in arcseconds')
+        parser.add_argument('--det', type=str, help='Detector to reduce. Same format as detnum')
+        parser.add_argument('--no_stack', dest='stack', default=True, 
+                            action="store_false",
+                            help='Do *not* stack multiple science frames')
+        return parser
+
+
+    @staticmethod
+    def main(args):
+
+        tstart = time.perf_counter()
+
+        # Ingest Files 
+        files = io.grab_rawfiles(
+            raw_paths=[args.full_rawpath], 
+            file_of_files=args.rawfile_list, 
+            list_of_files=args.rawfiles) 
+
+        # Run PypeIt Setup on all the files
+        ps = pypeitsetup.PypeItSetup.from_rawfiles(files,
+                                                args.spectrograph)
+        ps.run(setup_only=True, no_write_sorted=True)
+
+        # Calibrate, if necessary
+        calib_dir = args.calib_dir if args.calib_dir is not None else args.redux_path
+        if args.masters_dir is None:
+            # Generate PypeIt files (and folders)
+            calib_pypeit_files = ps.generate_ql_calib_pypeit_files(
+                calib_dir, det=args.det, configs='all',
+                clobber=args.clobber_calibs)
+            # Process them
+            for calib_pypeit_file in calib_pypeit_files: 
+                redux_path = os.path.dirname(calib_pypeit_file)  # Path to PypeIt file
+                # Check for existing masters
+                master_files = glob.glob(os.path.join(
+                    redux_path, 'Masters', 'Master*'))
+                if len(master_files) > 0 and not args.clobber_calibs:
+                    msgs.info('Master files already exist.  Skipping calibration.')
+                    continue
+                # Run
+                pypeIt = pypeit.PypeIt(calib_pypeit_file,
+                                       redux_path=redux_path, 
+                                       calib_only=True)
+                calib_dict = pypeIt.calib_all()
+        if args.calibs_only:
+            msgs.info("Calibrations only requested.  Exiting")
+            return
+
+        # Science files                                
+        if args.sci_files is not None:
+            sci_idx = np.in1d(ps.fitstbl['filename'], args.sci_files)
+        else:
+            sci_idx = ps.fitstbl['frametype'] == 'science'
```

Comment:
I think we should be using `fitstbl.find_frames('science')` specifically for IR cases where the frametype might be 'arc,science,tilt', not just 'science'.

Reply:
good one.  done